Modules/_sqlite/connection.c

```diff
@@ -1478,6 +1478,92 @@ pysqlite_connection_iterdump(pysqlite_Connection* self, PyObject* args)
 }
 
 static PyObject *
+pysqlite_connection_backup(pysqlite_Connection* self, PyObject* args, PyObject* kwds)
+{
+    char* filename;
+    int pages = -1;
+    PyObject* progress = Py_None;
+    PyObject* retval = NULL;
+    int rc;
+    sqlite3 *bckconn;
+    sqlite3_backup *bckhandle;
+    static char *keywords[] = {"filename", "pages", "progress", NULL};
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "s|$iO:backup", keywords,
+                                     &filename, &pages, &progress)) {
+        goto finally;
+    }
+
+    if (progress != Py_None && !PyCallable_Check(progress)) {
+        PyErr_SetString(PyExc_TypeError, "progress argument must be a callable");
+        goto finally;
+    }
+
+    if (pages == 0) {
+        pages = -1;
+    }
+
+    Py_BEGIN_ALLOW_THREADS
+    rc = sqlite3_open(filename, &bckconn);
+    Py_END_ALLOW_THREADS
+
+    if (rc != SQLITE_OK) {
+        goto finally;
+    }
+
+    Py_BEGIN_ALLOW_THREADS
+    bckhandle = sqlite3_backup_init(bckconn, "main", self->db, "main");
```

Comment:
Uhm, further investigation seems to reveal that the `dest_name` is pointless: _attached_ databases are **not** a persistent property of a database, so that we cannot allow the following:
```python
bck = sqlite.connect('/tmp/thebackup')
bck.execute(f"ATTACH DATABASE '/tmp/theattachment' AS attached_db")
bck.commit()

db = sqlite.connect('/tmp/database')
db.backup('/tmp/thebackup', dest_name='attached_db')
```

Changed back to a single `name` argument to specify the _source database_...